packages/sources/intrinio/src/util.ts

```diff
@@ -0,0 +1,296 @@
+        }
+      })
+
+      req.on('error', (e) => {
+        console.error('IntrinioRealtime | Could not get auth token: ' + e)
+        reject(e)
+      })
+
+      req.end()
+    })
+  }
+
+  async _makeSocketUrl(): Promise<string> {
+    if (!this.token) {
+      await this._refreshToken()
+      if (!this.token) return ''
+    }
+
+    if (this.options.provider == 'iex') {
+      return (
+        'wss://realtime.intrinio.com/socket/websocket?vsn=1.0.0&token=' +
+        encodeURIComponent(this.token)
+      )
+    } else if (this.options.provider == 'quodd') {
+      return 'wss://www5.quodd.com/websocket/webStreamer/intrinio/' + encodeURIComponent(this.token)
+    } else if (this.options.provider == 'cryptoquote') {
+      return (
+        'wss://crypto.intrinio.com/socket/websocket?vsn=1.0.0&token=' +
+        encodeURIComponent(this.token)
+      )
+    } else if (this.options.provider == 'fxcm') {
+      return (
+        'wss://fxcm.intrinio.com/socket/websocket?vsn=1.0.0&token=' + encodeURIComponent(this.token)
+      )
+    }
+
+    return ''
+  }
+
+  _makeHeartbeatMessage(): any {
+    if (this.options.provider == 'quodd') {
+      return { event: 'heartbeat', data: { action: 'heartbeat', ticker: Date.now() } }
+    } else if (['iex', 'cryptoquote', 'fxcm'].includes(this.options.provider)) {
+      return { topic: 'phoenix', event: 'heartbeat', payload: {}, ref: null }
+    }
+  }
+
+  _makeJoinMessage(channel: string): any {
+    if (this.options.provider == 'iex') {
+      return {
+        topic: this._parseIexTopic(channel),
+        event: 'phx_join',
+        payload: {},
+        ref: null,
+      }
+    } else if (this.options.provider == 'quodd') {
+      return {
+        event: 'subscribe',
+        data: {
+          ticker: channel,
+          action: 'subscribe',
+        },
+      }
+    } else if (['cryptoquote', 'fxcm'].includes(this.options.provider)) {
+      return {
+        topic: channel,
+        event: 'phx_join',
+        payload: {},
+        ref: null,
+      }
+    }
+  }
+
+  _makeLeaveMessage(channel: string): any {
+    if (this.options.provider == 'iex') {
+      return {
+        topic: this._parseIexTopic(channel),
+        event: 'phx_leave',
+        payload: {},
+        ref: null,
+      }
+    } else if (this.options.provider == 'quodd') {
+      return {
+        event: 'unsubscribe',
+        data: {
+          ticker: channel,
+          action: 'unsubscribe',
+        },
+      }
+    } else if (['cryptoquote', 'fxcm'].includes(this.options.provider)) {
+      return {
+        topic: channel,
+        event: 'phx_leave',
+        payload: {},
+        ref: null,
+      }
+    }
+  }
+
+  _parseIexTopic(channel: string): string {
+    let topic = null
+    if (channel == '$lobby') {
+      topic = 'iex:lobby'
+    } else if (channel == '$lobby_last_price') {
+      topic = 'iex:lobby:last_price'
+    } else {
+      topic = 'iex:securities:' + channel
+    }
+    return topic
```

Comment:
```suggestion
   switch (channel) {
 case '$lobby': 
  return 'iex:lobby'
case '$lobby_last_price': 
  return 'iex:lobby:last_price'
default:
  return `iex:securities:${channel}`
}
```

Good opportunity to use a switch/case statement here